src/utils.ts

```diff
@@ -317,7 +318,34 @@ export function isNonEmptyURL(url: string) {
  *
  * @group utils
  */
-export function joinURL(..._input: string[]): string {
+export function joinURL(base: string, ...input: string[]): string {
+  let url = base || "";
+
+  for (const segment of input.filter((url) => isNonEmptyURL(url))) {
+    if (url) {
+      // TODO: Handle .. when joining
```

Comment:
Do we still want to do this in `joinURL`?

Reply:
because as you know it is risky change to be in same semver major ++ there is no behavior we agree for (fs path, URL and what we proposed for ufo are inconsistent today).

Reply:
it is still todo and my intention to support only when confident that we do it properly.